docs/design/README.md

```diff
@@ -1,50 +1,63 @@
-# TiDB Design Documents
+# TiDB Improvement Proposals
```

Comment:
`improvement` is ambitious to cover performance improvements, new features

Reply:
I don't have a strong feeling on the name, either "RFCs" / "Design Documents" / "Improvement Proposals" is OK. Developers usually use them interchangeably. But given this file is under `docs/design `, I agree to change all mentions as "Design Documents".